tokio/src/net/pipe.rs

```diff
@@ -144,7 +146,7 @@ impl Sender {
             .custom_flags(libc::O_NONBLOCK)
             .open(path)?;
         let raw_fd = file.into_raw_fd();
-        // Safety: Raw fd was created a valid file.
+        // Safety: Raw fd was created from a valid file.
         let mio_tx = unsafe { mio_pipe::Sender::from_raw_fd(raw_fd) };
         Sender::from_mio(mio_tx)
```

Comment:
I think it's better to call `is_fifo` to verify that the `file` is indeed a pipe, then call `Sender::from_file_unchecked` to avoid the `unsafe` code.